### PR TITLE
Refactor calibration workers and fix metadata writing

### LINKED ISSUE #189: [BUG] The calibration_metadata has empty "Microscope parameters" field after calibration
The calibration (specifically MM-Voltage mode) produces a calibration_metadata file with  "Microscope parameters" set to null even though the correct parameters are set in Acquisition/Reconstruction -> Microscope Parameters tab. 

Another field by the same name is created after background is acquired and it is populated with correct parameters.
image
```
{
 "Summary": {
  "Timestamp": "2022-09-21 16:55:36.574805",
  "recOrder-napari version": "0.2.0rc2",
  "waveorder version": "1.0.0rc0"
 },
 "Calibration": {
  "Calibration scheme": "4-State",
  "Swing (waves)": 0.1,
  "Wavelength (nm)": 532,
  "Retardance to voltage interpolation method": "schnoor_fit",
  "LC control mode": "MM-Voltage",
  "Black level": 101.57,
  "Extinction ratio": 203.84,
  "ROI (x, y, width, height)": [
   0,
   0,
   2048,
   2048
  ],
  "Channel names": [
   "State0",
   "State1",
   "State2",
   "State3"
  ],
  "LC retardance": {
   "LCA_ext": 0.273487,
   "LCB_ext": 0.534724,
   "LCA_0": 0.173487,
   "LCB_0": 0.534724,
   "LCA_60": 0.322006,
   "LCB_60": 0.44773,
   "LCA_120": 0.319937,
   "LCB_120": 0.618008
  },
  "LC voltage": {
   "LCA_ext": 6.6239,
   "LCB_ext": 4.196,
   "LCA_0": 9.4735,
   "LCB_0": 4.196,
   "LCA_60": 5.8945,
   "LCB_60": 4.7117,
   "LCA_120": 5.9212,
   "LCB_120": 3.8246
  },
  "Swing_0": 0.1,
  "Swing_60": 0.1,
  "Swing_120": 0.095,
  "Instrument matrix": [
   [
    1.0,
    0.0,
    0.0,
    -1.0
   ],
   [
    1.0,
    0.58779,
    0.0,
    -0.80902
   ],
   [
    1.0,
    -0.29389,
    0.50904,
    -0.80902
   ],
   [
    1.0,
    -0.29389,
    -0.50904,
    -0.80902
   ]
  ]
 },
 "Notes": "test",
 "Microscope parameters": null
}
```
**Environment:**
Hummingbird (windows) recOrder-0.2.0-rc2.

### PR DESCRIPTION
**Pull request by ziw-liu**
This PR starts to refactor `recOrder.plugin.workers.calibration_workers` to reduce redundancy. Related: #170.

Fixes #189.

### CODE REVIEW

File: recOrder/plugin/workers/calibration_workers.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# TODO: remove in Python 3.11
+from __future__ import annotations
+
```
Comment (talonchandler) [inline recOrder/plugin/workers/calibration_workers.py:3]: Needed?

File: recOrder/plugin/workers/calibration_workers.py [hunk 2]
```diff
@@ -12,6 +15,13 @@
 import logging
 import json
 
+# type hint/check
+from typing import TYPE_CHECKING
+# avoid circular import error
+if TYPE_CHECKING:
+    from recOrder.plugin.widget.main_widget import MainWidget
+    from recOrder.calib.Calibration import QLIPP_Calibration
+
```
Comment (talonchandler) [inline recOrder/plugin/workers/calibration_workers.py:24]: Can you help me understand this?

**Comment by talonchandler** [comment]
I tested this branch and found it was double writing "Microscope parameters" and "Microscope Parameters" fields, and I fixed it by making a small change to `MetadataReader`.

Other than the new import questions, LGTM!

**Review by talonchandler — APPROVED** [review]

File: recOrder/plugin/workers/calibration_workers.py (hunk 1 above)
Comment (ziw-liu) [reply recOrder/plugin/workers/calibration_workers.py:3]: Doesn't change behavior. But some type hints will have to be strings without this, which is aesthetically suboptimal.

File: recOrder/plugin/workers/calibration_workers.py (hunk 2 above)
Comment (ziw-liu) [reply recOrder/plugin/workers/calibration_workers.py:24]: `main_widget` imports from `calibration_worker` and causes circular import issues in the base-class declaration. This avoids errors by only executing during type checking calls, which currently never happen in runtime. This is both for readability (native type hints) and being future-proof (if we will be using `pydantic` in 1.0.0+).

**Pull request merged**